### PR TITLE
Adding 4 new Reaction delete routes and obsoleting existing routes

### LINKED ISSUE #2490: Add way of deleting reactions from individual routes
In an older version of the api, there used to be a single route for deleting reactions. This has now changed into 4 individual routes for commit comments, issues, issue comments and pull request reviews. We should:

* Add a delete route (and tests) to ICommitCommentReactionsClient
* Add a delete route (and tests) to IIssueReactionsClient
* Add a delete route (and tests) to IIssueCommentReactionsClient
* Add a delete route (and tests) to IPullRequestReviewCommentReactionsClient
* Remove the old way
* Remove any excess tests

### PR DESCRIPTION
**Pull request by JonruAlveus**
fixes #2490 

#2430

### CODE REVIEW

File: Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs [hunk 1]
```diff
@@ -70,5 +71,26 @@ public interface IObservableCommitCommentReactionsClient
...
+        /// <summary>
+        /// Deletes a reaction for a specified Commit Comment
+        /// </summary>
+        /// <remarks>https://docs.github.com/en/rest/reactions#delete-a-commit-comment-reaction</remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="number">The comment id</param>
+        /// <param name="reaction">The reaction id</param>
+        /// <returns></returns>
+        IObservable<Unit> Delete(string owner, string name, int number, int reaction);
```
Comment (nickfloyd) [inline Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs:84]: For generation sake (future concern) and ease of consumption, I'd like to name these to match the names in the OpenAPI descriptions as closely as possible.

```suggestion
        /// <param name="commentId">The comment id</param>
        /// <param name="reactionId">The reaction id</param>
        /// <returns></returns>
        IObservable<Unit> Delete(string owner, string name, int commentId, int reactionId);
```

File: Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs [hunk 2]
```diff
@@ -70,5 +71,26 @@ public interface IObservableCommitCommentReactionsClient
...
+
+        /// <summary>
+        /// Deletes a reaction for a specified Commit Comment
+        /// </summary>
+        /// <remarks>https://docs.github.com/en/rest/reactions#delete-a-commit-comment-reaction</remarks>
+        /// <param name="repositoryId">The Id of the repository</param>
+        /// <param name="number">The comment id</param>
+        /// <param name="reaction">The reaction id</param>
+        /// <returns></returns>
+        IObservable<Unit> Delete(long repositoryId, int number, int reaction);
```
Comment (nickfloyd) [inline Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs:94]:
```suggestion
        /// <param name="commentId">The comment id</param>
        /// <param name="reactionId">The reaction id</param>
        /// <returns></returns>
        IObservable<Unit> Delete(long repositoryId, int commentId, int reactionId);
```

File: Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs [hunk 3]
```diff
@@ -64,5 +65,26 @@ public interface IObservableIssueCommentReactionsClient
...
+        /// <summary>
+        /// Deletes a reaction for a specified Issue Comment
+        /// </summary>
+        /// <remarks>https://docs.github.com/en/rest/reactions#delete-an-issue-comment-reaction</remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="number">The issue id</param>
+        /// <param name="reaction">The reaction id</param>
+        /// <returns></returns>
+        IObservable<Unit> Delete(string owner, string name, int number, int reaction);
```
Comment (nickfloyd) [inline Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs:78]:
```suggestion
        /// <param name="issueId">The issue id</param>
        /// <param name="reactionId">The reaction id</param>
        /// <returns></returns>
        IObservable<Unit> Delete(string owner, string name, int issueId, int reactionId);
```

File: Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs [hunk 4]
```diff
@@ -64,5 +65,26 @@ public interface IObservableIssueCommentReactionsClient
...
+
+        /// <summary>
+        /// Deletes a reaction for a specified Commit Comment
+        /// </summary>
+        /// <remarks>https://docs.github.com/en/rest/reactions#delete-an-issue-comment-reaction</remarks>
+        /// <param name="repositoryId">The Id of the repository</param>
+        /// <param name="number">The comment id</param>
+        /// <param name="reaction">The reaction id</param>
+        /// <returns></returns>
+        IObservable<Unit> Delete(long repositoryId, int number, int reaction);
```
Comment (nickfloyd) [inline Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs:88]:
```suggestion
        /// <param name="commentId">The comment id</param>
        /// <param name="reactionId">The reaction id</param>
        /// <returns></returns>
        IObservable<Unit> Delete(long repositoryId, int commentId, int reactionId);
```

**Review by nickfloyd — CHANGES_REQUESTED** [review]
Hey @JonruAlveus thanks for the contributions here.  I just commented on the first few files regarding some parameter naming - let me know your thoughts; the changes would need to be applied through the entire change set.

File: Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs (hunk 1 above)
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs]: 👍

File: Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs (hunk 2 above)
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/IObservableCommitCommentReactionsClient.cs]: 👍

File: Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs (hunk 3 above)
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs]: 👍

File: Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs (hunk 4 above)
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/IObservableIssueCommentReactionsClient.cs]: 👍

**Comment by JonruAlveus** [comment]
@nickfloyd all changes made!

**Review by nickfloyd — APPROVED** [review]
Thanks for the changes here! I was looking at Octokit/Helpers/ApiUrls.cs, and it looks like that whole class needs to be refactored to fit a less obscure naming scheme - lots of use of the nondescript `number` and the like.  

I'll create an issue for that one so that we can move forward with your changes here.

**Comment by nickfloyd** [comment]
release_notes: Adds 4 new Reaction delete routes to replace the existing obsolete routes

**Pull request merged**